test/parallel/test-timers-get-active-timers.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { getActiveTimers } = require('timers');
+
+const inputs = [
+  1,
+  2,
+  35,
+  47,
+  47,
+  200,
+  200,
+  256,
+];
+
+// Setting timeouts and intervals using values from the inputs array.
+// These timers should not be called.
+inputs.forEach((value) => setTimeout(common.mustNotCall(), value));
+inputs.forEach((value) => setInterval(common.mustNotCall(), value));
+
+// Setting timers that should not be showing up when
+// we use clearTimeout before calling getActiveTimers.
+const clearedTimers = inputs.map((value) =>
+  setTimeout(common.mustNotCall(), value));
+
+clearedTimers.forEach(clearTimeout);
+
+const activeTimers = getActiveTimers();
+
```

Comment:
Could you loop over all of the timeouts/intervals created above, and verify that each one is in `activeTimers` (that would have the added benefit of verifying that the array actually contains timeout objects). Then, loop over all of the `clearedTimers` and assert that none of them are in `activeTimers`.

Reply:
This is what I was thinking too. It should verify that the objects returns are the same ones made during scheduling.
The order is stable, so it shouldn't be too difficult to do I think.